test/integration/generators/test_generators_from_br.py

```diff
@@ -0,0 +1,22 @@
+import textwrap
+
+from conan.test.assets.genconanfile import GenConanfile
+from conan.test.utils.tools import TestClient
+
+
+def test_inject_generators_conf():
+    tc = TestClient(light=True)
+    tc.save({"tool/conanfile.py": textwrap.dedent("""
+    from conan import ConanFile
+    class ToolConan(ConanFile):
+        name = "tool"
+        version = "0.1"
+
+        def package_info(self):
+            self.generator_info = ["CMakeToolchain"]
```

Comment:
This is missing the important use case of injecting a ``MyGenerator`` class defined in this conanfile